integreat_cms/cms/models/media/media_file.py

```diff
@@ -115,6 +119,33 @@ def file_size_limit(value):
         )
 
 
+class MediaFileQuerySet(models.QuerySet):
+    """
+    Custom queryset for media files
+    """
+
+    def filter_unused(self):
+        r"""
+        Filter for unused media files
+
+        :return: The queryset of unused media files
+        :rtype: ~django.db.models.query.QuerySet [ ~integreat_cms.cms.models.media.media_file.MediaFile ]
+        """
+        urls = Url.objects.filter(
+            url=Concat(
+                Value(settings.BASE_URL), Value(settings.MEDIA_URL), OuterRef("file")
+            )
+        )
+        return self.annotate(is_embedded=Exists(urls)).filter(
+            icon_organizations__isnull=True,
+            icon_regions__isnull=True,
+            events__isnull=True,
+            pages__isnull=True,
+            pois__isnull=True,
+            is_embedded=False,
+        )
```

Comment:
This query finds all usages with one sql query - this will probably be much faster than querying all media objects from the database and checking whether they're used in a loop.

Reply:
It's so cool 😻 Thank you!